__tests__/supertest.js

```diff
@@ -0,0 +1,417 @@
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- planter_id should be a number `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('planter_id', 'number');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- planter_identifier is required `, function (done) {
+      const request_object = new RequestObject();
+      request_object.delete_property('planter_identifier');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- planter_identifier should be a string `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('planter_identifier', 500);
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- planter_photo_url should be a url `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('planter_photo_url', '5000');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- attributes should be an array `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('attributes', '5000');
+      request(server)
+        .post(`/raw-captures`)
+        .send(request_object.request_object)
+        .set('Accept', 'application/json')
+        .expect(422)
+        .end(function (err, res) {
+          if (err) return done(err);
+          return done();
+        });
+    });
+
+    it(`Should raise validation error with error code 422 -- attributes should be an array of objects `, function (done) {
+      const request_object = new RequestObject();
+      request_object.change_property('attributes', ['5000']);
```

Comment:
attributes is usually an array of objects with key/value pair like the following. What you have for this test case is fine, just wanted to let give you an example of  a valid attribute value.
``` 
[{
"key": "rotation_matrix",
"value": "0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0"
 },
{ "key": ....
  "value": ...
}
]
```

Reply:
Oh okay, I'll make changes to the schema, and also update the tests